uiowa/uiowa
# Feature:  Top links menu and character count limit for inline header

Resolves https://github.com/uiowa/uiowa/issues/2537 and resolves https://github.com/uiowa/uiowa/issues/1980

# How to test 

## How to test top links

1. `blt ds --site=admissions.uiowa.edu`
2. `blt ds --site=uiowa.edu`
3. `yarn workspace uids_base gulp --development`
4. Add 2 top links to http://admissions.local.drupal.uiowa.edu/admin/structure/menu/manage/top-links?destination=/admin/structure/menu
5. Save and view site. 
   - Test how menu adapts to all breakpoints
6. Change header navigation styles at http://admissions.local.drupal.uiowa.edu/admin/appearance/settings/uids_base
   -  Test how menu adapts to all breakpoints
7. Add parent organization and top links
   -  Test how menu adapts to all breakpoints
8. Test on `uiowa.edu` local site and test. 

## How to test character count limit

1. Add a very long site name over 43 characters to http://admissions.local.drupal.uiowa.edu/admin/config/system/site-information and navigate to http://admissions.local.drupal.uiowa.edu/admin/appearance/settings/uids_base and verify that the ability to set the site name display is grayed-out. 
2. Add a very short site name under 43 characters to http://admissions.local.drupal.uiowa.edu/admin/config/system/site-information and navigate to http://admissions.local.drupal.uiowa.edu/admin/appearance/settings/uids_base and verify that the ability to change site name display is available.
3. Save settings and verify that the correct header is displayed. 

## Theme settings changes and update hook
The following items are removed from `uids_base` config (which is entirely config ignored):
* `uids_base` - We used to namespace all items under this key
* `branding` - This was used to allow for the `legacy` setting, which has since been removed.

The update hook runs against all sites and removes those items.  This includes HR, which isn't using `uids_base` but still has config installed for it. This will ensure that whenever HR comes back into the fold, it will have up-to-date theme config. 

I tested it against:
* A new install - to make sure that the settings are installed correctly.
* hawkeyemarchingband.uiowa.edu - V2
* HR - Non-uids_base site
* uiowa.edu

To test, I ran `blt ds --site={site}` and then logged in and checked the config via https://default.local.drupal.uiowa.edu/admin/config/development/configuration/single/export. 

Feel free to check as many sites or potential edge cases as you feel is warranted.
